src/ajax.js

```diff
@@ -557,8 +557,14 @@ jQuery.extend( {
 			}
 		}
 
+		// Uppercase the type
+		s.type = s.type.toUpperCase();
+
+		// Determine if request has content
+		s.hasContent = !rnoContent.test( s.type );
+
 		// Convert data if not already a string
-		if ( s.data && s.processData && typeof s.data !== "string" ) {
+		if ( s.data && ( s.processData || !s.hasContent ) && typeof s.data !== "string" ) {
```

Comment:
@Litor I think I understand the problem you're running into now. With HTTP methods that do not have a request body such as GET, any provided `data` is [always processed](https://github.com/jquery/jquery/blob/4e50967725e61a5d71c2792131de61dec0520f9c/src/ajax.js#L594) even if the caller specified `processData: false`.  This seems contrary to the documentation which says:

> The data option can contain either a query string of the form key1=value1&key2=value2, or an object of the form {key1: 'value1', key2: 'value2'}. If the latter form is used, the data is converted into a query string using jQuery.param() before it is sent. This processing can be circumvented by setting processData to false.

This behavior was also present before 3.x so it's not any kind of recent regression, but it does seem like a bug. @gibson042 do you agree? If we didn't process the data it would be possible for a `beforeSend` handler to take an object and stringify it to the URL as desired like: http://jsbin.com/wixuquzeri/edit?html,console